### ISSUE TITLE
Remember lobby option checkbox states

### ISSUE THREAD

**Issue by xan2622** [OpenRA/OpenRA#4122]
I think it could be interesting if Map Options (shroud, fog of war, crates appear...) could be remembered after modificating them.

This way, a player could create new games according to his preferences (and without always having to re-enable them each time a new game is created).

Same for the Debug (Developer Mode), checkboxes could be saved too.

**Comment by abcdefg30** [comment]
Just in case someone wants to work on this:

> 21:19:01  (abcdefg30) you could fix #4122
21:19:14  (pchote) please don't
21:19:17  (abcdefg30) ?
21:19:23  (pchote) that issue is full of subtlety
21:20:16  (pchote) its bad enough when switching to a map with custom rules dumps out the default bots
21:20:58  (abcdefg30) and it doesn't change e.g. your faction if you change from a map with custom rules to a normal one...
21:21:11  (abcdefg30) well ok so fixing ^ is required before fixing #4122?
21:21:37  (pchote) the problem is when you switch to a map that (for example) force disables shroud, and then back to a normal map
21:21:46  (pchote) if you remembered the last state, then your new map will start with shroud disabled
21:21:53  (pchote) or starting units configured, or whatever
21:22:08  (pchote) which is something that the host had manually set
21:22:22  (pchote) and is asking for endless bug reports from people who are confused about why their settings changed
21:22:52  (pchote) *which is not something that was manually set
21:23:46  (hamb) +1
21:24:07  (abcdefg30) well, we shouldn't remember the settings of custom maps anyway
21:24:11  (abcdefg30) as i said ^
21:24:11  (Scott\`) that's lame
21:24:21  (Scott\`) we already solved that issue for player colours
21:24:23  (abcdefg30) you can get the "allied" faction from custom maps on a normal map 
21:24:39  (abcdefg30) hm, Scott\` what was the solution again? i don't remember
21:24:57  (Scott\`) have 2 colours, a colour preference and the actual colour
21:25:14  (Scott\`) map chooses the actual colour or lets you use your preference
21:25:23  (abcdefg30) ok, so same to settings?
21:25:26  (Scott\`) using the colour picker sets the colour preference
21:25:35  (abcdefg30) sounds good to me
21:26:56  (Scott\`) i even implemented ^ for the same reason
21:27:06  (Scott\`) i got annoyed that my colour wasnt persisting after switching to a mission map
21:27:08  (Scott\`) :P

**Comment by pchote** [comment]
My [lobby-trait-options](https://github.com/pchote/OpenRA/commits/lobby-trait-options) branch will fix this.

**Comment by rob-v** [comment]
I implemented recently settings per mod #4787 so I looked also at remembering other options as requested here. I added 'Remember options' settings so it would be optional, but there is one issue - Maps have their default/preferred values. When we use stored lobby options, then map default (recommended?) options would never be displayed/used.

**Comment by BGluth** [comment]
Not sure whether to post here or in #4787...

@pchote 

I'm not sure if this is still useful, but I managed to get storing per session lobby settings (eg. kill bounties) and occupied slots data for Skirmish/MP games (#15381) going. Everything involving slots is being done with lobby commands and testing over a local network it seems to be behaving correctly. However, I'm pretty sure storing the settings to some mod specific yaml is probably the better way to go (which I didn't do), but I guess that it can maybe be modified to do that instead when it's later available (from the discussion in #12990).

Is it still worthwhile to submit a PR for this?

Here's the commit message
```
Lobby settings are now saved when the lobby window is closed

Saved by the admin:
    - Each's bots slot, faction, team and spawn
    - Game settings (kill bounties, crates, starting cash, etc.)

Saved by the client:
    - Faction, team, and spawn

- When a lobby is created, any stored game specific settings are applied ONLY by the host.
- Settings are saved when the lobby window is disposed.
- When a map uses custom rules saving/reapplying settings is disabled.
- Clients connecting to a game that they are not hosting will not apply their stored team and spawn and only the faction.
- All settings are stored only per session and are not available on disk.
- All settings for lobby slots are applied with lobby commands.
- Settings applied to a local skirmish game will be applied to a MP game if the client hosts one.
```
